src/guest.S

```diff
@@ -218,6 +227,14 @@ _guest_exit:
     sd    t2, ({guest_fcsr})(a0)
     csrw  sstatus, t1
 
+_save_guest_vector_state:
+    /* Save vector state if needed     */
+    /* We do not have a stack. Save    */ 
+    /* ra in t6 instead of on the stack*/
+    mv t6, ra
+    call _guest_exit_vector
+    mv ra, t6
```

Comment:
This is a bit hairy. We don't actually need to save/restore vector state in the `_run_guest` path since Salus isn't using vectors and thus it won't affect its execution environment. So we could have proper function calls for saving/restoring vectors. Maybe would look something like this:

```
pub fn run_to_exit(&mut self) -> VmCpuExit {
    ...
    self.save_vectors();
    unsafe { _run_guest(...); }
    self.restore_vectors();
    ```
}

#[cfg(...)]
fn save_vectors(&mut self) {
    if self.state.guest_regs.sstatus.read(sstatus::vs) == sstatus::vs::Dirty {
        _save_vectors(...); // <- asm function to save vector state
        self.state.guest_regs.sstatus.modify(sstatus::vs::Clean)
    }
}

#[cfg(not(...)]
fn save_vectors(&mut self) {}

#[cfg(...)]
fn restore_vectors(&mut self) {
    _restore_vectors(...); // <- asm function to restore vector state
}

#[cfg(not(..))]
fn restore_vectors(&mut self) {}
```

This would also let you get rid of the dummy asm file since we'd do the #[cfg] selection around the Rust function call.

Also I realize FP state could (and probably should) do something similar. I can send a PR for that so that we end up being consistent for both FP and vector state.